src/graph/entities/attribute_set.c

```diff
@@ -284,12 +340,21 @@ bool AttributeSet_Update
 	ASSERT(set != NULL);
 	ASSERT(attr_id != ATTRIBUTE_ID_NONE);
 
+	AttributeSet _set = *set;
+
+	// return if set is read-only
+	if(unlikely(ATTRIBUTE_SET_IS_READONLY(_set))) {
+		return false;
+	}
+
+	ASSERT(_set != NULL);
```

Comment:
remove assert

Reply:
going to keep it, it seems like the only place which calls this function make sure the updated attribute exists, also I think update should acts on an existing attribute, we can't update a none existing attribute